src/instructlab/utils.py

```diff
@@ -605,3 +605,39 @@ def ensure_legacy_dataset(
         return dataset  # type: ignore
 
     return convert_messages_to_legacy_dataset(dataset)  # type: ignore
+
+
+def is_oci_repo(repo_url: str) -> bool:
+    """
+    Checks if a provided repository follows the OCI registry URL syntax
+    """
+
+    # TODO: flesh this out and make it a more robust check
+    oci_url_prefix = "docker://"
+    return repo_url.startswith(oci_url_prefix)
+
+
+def is_huggingface_repo(repo_name: str) -> bool:
+    # allow alphanumerics, underscores, hyphens and periods in huggingface repo names
+    # repo name should be of the format <owner>/<model>
+    pattern = r"^[\w.-]+\/[\w.-]+$"
+    return re.match(pattern, repo_name) is not None
+
+
+def _load_json(file_path: Path):
```

Comment:
(No action required) These functions seem closely coupled to download module. (The sha one also to OCI implementation.) I'd put them back into `download.py`.